### PR TITLE
Release 0.4.1

### PR DESCRIPTION
**Pull request by JPXKQX**
Release 0.4.1 of `anemoi.models`

This PR includes the latest developments:

- #37 
- #56 
- #89 
- #97 
- #88 
- #102 

----
📚 Documentation preview 📚: https://anemoi-models--94.org.readthedocs.build/en/94/

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-models?pullRequest=94) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-models?pullRequest=94) before we can accept your contribution.<br/>**6** out of **7** committers have signed the CLA.<br/><br/>:white_check_mark: cathalobrien<br/>:white_check_mark: sahahner<br/>:white_check_mark: mchantry<br/>:white_check_mark: JPXKQX<br/>:white_check_mark: icedoom888<br/>:white_check_mark: b8raoult<br/>:x: dietervdb-meteo<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/anemoi-models?pullRequest=94) it.</sub>

File: CHANGELOG.md
Comment (anaprietonem) [inline CHANGELOG.md:11]: We would need this to be the header for 0.4.1 don't we?

File: src/anemoi/models/models/hierarchical.py
Comment (anaprietonem) [inline src/anemoi/models/models/hierarchical.py:9]: Great great, and nice to have this feature added! I would just like to point out that seems like a new feature that could have used some documentation. Was that added during the PR ? If not probably we could at least create a new PR to include an update to the docs

**Review by anaprietonem — COMMENTED** [review]
Left some comments regarding Changelog and docs. I think that both, the Dynamic Mixin for the imputer and the Hierarchical Graph could include some update to the documentation. Maybe this already happened at the PR level, and the link is missing? If we don't want to block the release I would still acknowledge that with an issue and try to update then with a PR

File: src/anemoi/models/models/hierarchical.py
Comment (anaprietonem) [reply src/anemoi/models/models/hierarchical.py:9]: Comment addressed on this `[PR](https://github.com/ecmwf/anemoi-models/pull/94/commits/225315ea8e4cac6480e89cbb3bbd73c543386614 )`

**Review by anaprietonem — APPROVED** [review]
Previous comments regarding documentation and changelog were addressed so happy to approve the PR

**Comment by mchantry** [comment]
#88 should go into this release.

**Review by anaprietonem — APPROVED** [review]

**Pull request merged**
